charts/redpanda/chart_test.go

```diff
@@ -143,90 +140,6 @@ func TestChart(t *testing.T) {
-					// really care about the stability of their output, so globally
-					// disable them.
-					"tests.enabled=false",
-					// jwtSecret defaults to a random string. Can't have that
-					// in snapshot testing so set it to a static value.
-					"console.secret.login.jwtSecret=SECRETKEY",
-				},
-			})
-			require.NoError(t, err)
-
-			oldRedpanda, oldRPKProfile, err := getConfigMaps(manifests)
-			require.NoError(t, err)
-
-			// Now helm template will generate Redpanda configuration from local definition
-			manifests, err = client.Template(ctx, ".", helm.TemplateOptions{
-				Name:       "redpanda",
-				ValuesFile: "./ci/" + v.Name(),
-				Set: []string{
-					// Tests utilize some non-deterministic helpers (rng). We don't
-					// really care about the stability of their output, so globally
-					// disable them.
-					"tests.enabled=false",
-					// jwtSecret defaults to a random string. Can't have that
-					// in snapshot testing so set it to a static value.
-					"console.secret.login.jwtSecret=SECRETKEY",
-				},
-			})
-			require.NoError(t, err)
-
-			newRedpanda, newRPKProfile, err := getConfigMaps(manifests)
-			require.NoError(t, err)
-
-			// Overprovisioned field till Redpanda chart version 5.8.8 was wrongly set to `false`
-			// when CPU request value was bellow 1000 mili cores. Function `redpanda-smp`, that
-			// should overwrite `overprovisioned` flag in old implementation, was not called
-			// before setting `overprovisioned` flag (`{{ dig "cpu" "overprovisioned" false .Values.resources }}`).
-			// redpanda-smp template - https://github.com/redpanda-data/helm-charts/blob/5f287d45a3bda2763896840e505fb3de82b968b6/charts/redpanda/templates/_helpers.tpl#L187
-			// redpanda-smp template invocation - https://github.com/redpanda-data/helm-charts/blob/5f287d45a3bda2763896840e505fb3de82b968b6/charts/redpanda/templates/_configmap.tpl#L610
-			// overprovisioned flag - https://github.com/redpanda-data/helm-charts/blob/5f287d45a3bda2763896840e505fb3de82b968b6/charts/redpanda/templates/_configmap.tpl#L607
-			var newUnstructuredRedpandaConf map[string]any
-			require.NoError(t, yaml.Unmarshal([]byte(newRedpanda.Data["redpanda.yaml"]), &newUnstructuredRedpandaConf))
-			require.NoError(t, unstructured.SetNestedField(newUnstructuredRedpandaConf, false, "rpk", "overprovisioned"))
-
-			require.Equal(t, getJSONObject(t, oldRedpanda.Data["redpanda.yaml"]), newUnstructuredRedpandaConf)
-			require.Equal(t, getJSONObject(t, oldRedpanda.Data["bootstrap.yaml"]), getJSONObject(t, newRedpanda.Data["bootstrap.yaml"]))
-			require.Equal(t, getJSONObject(t, oldRPKProfile.Data["profile"]), getJSONObject(t, newRPKProfile.Data["profile"]))
-		})
-	}
-}
-
-func getJSONObject(t *testing.T, input string) any {
-	var output any
-	require.NoError(t, yaml.Unmarshal([]byte(input), &output))
-	return output
-}
-
 // getConfigMaps is parsing all manifests (resources) created by helm template
```

Comment:
NIT: `getConfigMaps` is unused now. It could be removed

Reply:
`CertTrustStoreCases` uses this